rllib/examples/curiosity/inverse_dynamics_model_based_curiosity.py

```diff
@@ -209,7 +218,37 @@ def on_sample_end(
             num_sgd_iter=6,
             lr=0.0003,
         )
-        .rl_module(model_config_dict={"vf_share_layers": True})
+        .rl_module(
+            rl_module_spec=MultiRLModuleSpec(
+                module_specs={
+                    # The "main" RLModule (policy) to be trained by our algo.
+                    DEFAULT_MODULE_ID: RLModuleSpec(
+                        model_config_dict={"vf_share_layers": True},
+                    ),
+                    # The inverse dynamics model.
+                    ICM_MODULE_ID: RLModuleSpec(
+                        module_class=InverseDynamicsModel,
+                        # Only create the ICM on the Learner workers, NOT on the
+                        # EnvRunners.
+                        learner_only=True,
```

Comment:
Ah I get it there are modules that have nothing which would be inference-only. Therefore they should not even be build in inference. Still these double flags are maybe not the best way to put it. It is a bit confusing - at least at the beginning.

Reply:
I get your confusion. Yes, these two flags are not complementary (inference_only=False does not mean learner_only=True or vice-versa).

I can try to describe this better in the docstrings. Maybe along the lines of:
* `inference_only`: Users have chance to streamline their custom RLModules by providing an `inference_only` mode that is then used by RLlib on EnvRunners. `inference_only` is NOT a config setting that users should need to set (or unset). Only RLlib should decide when an RLModule should be built in its `inference_only=True|False` state. The only exception is maybe when a user wants to do inference in production and loads the module from a checkpoint (with `inference_only=True` to save memory).
* `learner_only`: This is a setting that a user can set to indicate to RLlib that the RLModule will only be used on the Learners and should never be built on any EnvRunners.